src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -4307,48 +4339,89 @@ namespace FanCoilUnits {
                 if (PLR > 1.0) PLR = 1.0;
-                // adjust the PLR to meet the heating load by calling Calc4PipeFanCoil repeatedly
-                while (std::abs(Error) > ControlOffset && std::abs(AbsError) > Small5WLoad && Iter < MaxIterCycl && PLR != 1.0) {
-                    mdot = PLR * FanCoil(FanCoilNum).MaxHeatCoilFluidFlow;
-                    Node(InletNode).MassFlowRateMinAvail = Node(InletNode).MassFlowRate;
-                    SetComponentFlowRate(mdot,
-                                         FanCoil(FanCoilNum).HeatCoilFluidInletNode,
-                                         FanCoil(FanCoilNum).HeatCoilFluidOutletNodeNum,
-                                         FanCoil(FanCoilNum).HeatCoilLoopNum,
-                                         FanCoil(FanCoilNum).HeatCoilLoopSide,
-                                         FanCoil(FanCoilNum).HeatCoilBranchNum,
-                                         FanCoil(FanCoilNum).HeatCoilCompNum);
+                if (FanCoil(FanCoilNum).HCoilType_Num == HCoil_Water) {
+                    // adjust the PLR to meet the heating load by calling Calc4PipeFanCoil repeatedly
+                    while (std::abs(Error) > ControlOffset && std::abs(AbsError) > Small5WLoad && Iter < MaxIterCycl && PLR != 1.0) {
+                        Node(InletNode).MassFlowRateMinAvail = Node(InletNode).MassFlowRate;
+                        mdot = PLR * FanCoil(FanCoilNum).MaxHeatCoilFluidFlow;
+                        SetComponentFlowRate(mdot,
+                                             FanCoil(FanCoilNum).HeatCoilFluidInletNode,
+                                             FanCoil(FanCoilNum).HeatCoilFluidOutletNodeNum,
+                                             FanCoil(FanCoilNum).HeatCoilLoopNum,
+                                             FanCoil(FanCoilNum).HeatCoilLoopSide,
+                                             FanCoil(FanCoilNum).HeatCoilBranchNum,
+                                             FanCoil(FanCoilNum).HeatCoilCompNum);
+                        if (FanCoil(FanCoilNum).FanOpMode == ContFanCycCoil) {
+                            Calc4PipeFanCoil(state, FanCoilNum, ZoneNum, FirstHVACIteration, QUnitOut);
+                        } else {
+                            Calc4PipeFanCoil(state, FanCoilNum, ZoneNum, FirstHVACIteration, QUnitOut, PLR);
+                        }
+                        Error = (QZnReq - QUnitOut) / QZnReq;
+                        AbsError = QZnReq - QUnitOut;
+                        DelPLR = (QZnReq - QUnitOut) / QUnitOutMax;
+                        PLR += Relax * DelPLR;
+                        PLR = max(0.0, min(1.0, PLR));
+                        ++Iter;
+                        if (Iter == 32) Relax = 0.5;
+                        if (Iter == 65) Relax = 0.25;
+                        if (Iter > 70 && PLR == 0.0 && DelPLR < 0.0) Error = 0.0; // exit loop if PLR = 0
+                    }
```

Comment:
This is an existing code section for hot water heating coil. Just added if-logic to encapsulate all the hot water heating coil calculations together and the else below handles the electric heating coil.